pkg/deploy/controller/deployerpod/controller.go

```diff
@@ -50,6 +56,12 @@ func (c *DeployerPodController) Handle(pod *kapi.Pod) error {
 		}
 	case kapi.PodFailed:
 		nextStatus = deployapi.DeploymentStatusFailed
+		// if the deployment is already marked Failed, do not attempt clean up again
+		if currentStatus != nextStatus {
+			// failure to clean up will be logged but the deployment will still be marked Failed
+			err := c.cleanupFailedDeployment(deployment)
```

Comment:
We can mark failed until cleanup completes, because the rc may not get set.  Needs to retry set failed as long as scale fails